tasks/backfill_gh_app_installations.py

```diff
@@ -0,0 +1,198 @@
+    ):
+        print("backfill existing apps")
+        # Get owners that have installations, and installations queries
+        owners_query = (
+            db_session.query(Owner)
+            .join(GithubAppInstallation, Owner.ownerid == GithubAppInstallation.ownerid)
+            .filter(
+                Owner.service == "github",
+            )
+        )
+        gh_app_installations_query = db_session.query(GithubAppInstallation)
+
+        # Filter if owner_ids were provided
+        if owner_ids:
+            owners_query = owners_query.filter(Owner.ownerid.in_(owner_ids))
+            gh_app_installations_query = gh_app_installations_query.filter(
+                GithubAppInstallation.ownerid.in_(owner_ids)
+            )
+
+        owners: List[Owner] = owners_query.all()
+        gh_app_installations: List[
+            GithubAppInstallation
+        ] = gh_app_installations_query.all()
+
+        # I need to make reference of these in the gh installation app, Ill take suggestions here
+        owners_dict = {owner.ownerid: owner for owner in owners}
+
+        for gh_app_installation in gh_app_installations:
+            # Check if gh app has 'all' repositories selected
+            installation_id = gh_app_installation.installation_id
+
+            owner = owners_dict[gh_app_installation.ownerid]
+            ownerid = owner.ownerid
+
+            owner_service = get_owner_provider_service(
+                owner=owner, using_integration=True
+            )
+
+            remote_gh_app_installation = await owner_service.get_gh_app_installation(
+                installation_id=installation_id
+            )
+            repository_selection = remote_gh_app_installation.get(
+                "repository_selection", ""
+            )
+            if repository_selection == "all":
+                gh_app_installation.repository_service_ids = None
+                db_session.commit()
+                log.info(
+                    "Selection is set to all, no installation is needed",
+                    extra=dict(ownerid=ownerid),
+                )
+            else:
+                # Find and add all repos the gh app has access to
+                await self.add_repos_service_ids_from_provider(
+                    db_session=db_session,
+                    ownerid=ownerid,
+                    owner_service=owner_service,
+                    gh_app_installation=gh_app_installation,
+                )
+                log.info("Successful backfill", extra=dict(ownerid=ownerid))
+
+    async def backfill_owners_with_integration_without_gh_app(
+        self, db_session: Session, owner_ids: List[int] = None
+    ):
+        print("backfill new apps")
+        owners_with_integration_id_without_gh_app_query = (
+            db_session.query(Owner)
+            .join(
+                GithubAppInstallation,
+                Owner.ownerid != GithubAppInstallation.ownerid,
```

Comment:
This seems like it would be
```sql
select * from owners join GithubAppInstallation on owner.ownerid != GithubAppInstallation.ownerid
```
That would return all of the installations for other owners. 

I assume the goal is to find owners without the installation. If so you could use a left join and where null.
```sql
select * from owners left join GithubAppInstallation on owners.ownerid = GithubAppInstallation.owner_id where  GithubAppInstallation.owner_id is null
```

Reply:
I think what we want is this slightly modified query from yours, 

``` sql
select * from owners left join GithubAppInstallation on owners.ownerid = GithubAppInstallation.owner_id where  GithubAppInstallation.owner_id is null and owner.integration_id is not null
```

Difference being to filter by owner.integration_id != null, which went from "owners without gh app" to "owners without gh app with an installation id", cause we don't want to backfill for customers without an integration id to start with